src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -109,24 +108,28 @@ String metricName(MetricName metricName) {
         return prefix + '_' + group + '_' + name;
     }
 
-    static MetricFamilySamples convert(String name, String help, KafkaMetric metric, Map<String, String> labels) {
-        Object value = metric.metricValue();
-        if (!(value instanceof Number)) {
-            // Prometheus only accepts numeric metrics.
-            // Kafka gauges can have arbitrary types, so skip them for now
-            // TODO move non-numeric values to labels
-            return null;
-        }
-        Map<String, String> sanitizedLabels = labels.entrySet().stream()
+    private static MetricFamilySamples convert(String prometheusMetricName, KafkaMetric metric, MetricName metricName) {
+        Map<String, String> sanitizedLabels = metricName.tags().entrySet().stream()
                 .collect(Collectors.toMap(
                         e -> Collector.sanitizeMetricName(e.getKey()),
                         Map.Entry::getValue,
                         (v1, v2) -> {
-                            throw new IllegalStateException("Unexpected duplicate key " + v1);
+                            LOG.warn("Ignoring metric value duplicate key {}", v1);
+                            return v1;
                         },
                         LinkedHashMap::new));
-        return new MetricFamilySamplesBuilder(Type.GAUGE, help)
-                .addSample(name, ((Number) value).doubleValue(), sanitizedLabels)
+        Object valueObj = metric.metricValue();
+        double value;
+        if (valueObj instanceof Number) {
+            value = ((Number) valueObj).doubleValue();
+        } else {
+            value = 1.0;
+            String attributeName = metricName.name();
+            sanitizedLabels.put(attributeName, String.valueOf(valueObj));
```

Comment:
We probably want to sanitize the new label too.

Reply:
@mimaison Does this suffice in sanitizing the new label?
` sanitizedLabels.put(Collector.sanitizeMetricName(attributeName), String.valueOf(valueObj));`

Reply:
Yes let's use the same method we use on the other labels

Reply:
`Collector.sanitizeMetricName` used